posthog/models/integration.py

```diff
@@ -1138,6 +1163,268 @@ def create_issue(self, config: dict[str, str]):
+            }
+
+    def get_branch_info(self, repository: str, branch_name: str) -> dict[str, Any]:
+        """Get information about a specific branch."""
+        org = self.organization()
+        access_token = self.integration.sensitive_config["access_token"]
+
+        response = requests.get(
+            f"https://api.github.com/repos/{org}/{repository}/branches/{branch_name}",
+            headers={
+                "Accept": "application/vnd.github+json",
+                "Authorization": f"Bearer {access_token}",
+                "X-GitHub-Api-Version": "2022-11-28",
+            },
+        )
+
+        if response.status_code == 200:
+            branch_data = response.json()
+            return {
+                "success": True,
+                "exists": True,
+                "branch_name": branch_data["name"],
+                "commit_sha": branch_data["commit"]["sha"],
+                "protected": branch_data.get("protected", False),
+            }
+        elif response.status_code == 404:
+            return {
+                "success": True,
+                "exists": False,
+                "branch_name": branch_name,
+            }
+        else:
+            return {
+                "success": False,
+                "error": f"Failed to get branch info: {response.text}",
+                "status_code": response.status_code,
+            }
+
+    def list_pull_requests(self, repository: str, state: str = "open") -> dict[str, Any]:
+        """List pull requests for a repository."""
+        org = self.organization()
+        access_token = self.integration.sensitive_config["access_token"]
+
+        response = requests.get(
+            f"https://api.github.com/repos/{org}/{repository}/pulls",
+            params={"state": state, "per_page": 100},
+            headers={
+                "Accept": "application/vnd.github+json",
+                "Authorization": f"Bearer {access_token}",
+                "X-GitHub-Api-Version": "2022-11-28",
+            },
+        )
+
+        if response.status_code == 200:
+            prs = response.json()
+            return {
+                "success": True,
+                "pull_requests": [
+                    {
+                        "number": pr["number"],
+                        "title": pr["title"],
+                        "url": pr["html_url"],
+                        "state": pr["state"],
+                        "head_branch": pr["head"]["ref"],
+                        "base_branch": pr["base"]["ref"],
+                        "created_at": pr["created_at"],
+                        "updated_at": pr["updated_at"],
+                    }
+                    for pr in prs
+                ],
+            }
+        else:
+            return {
+                "success": False,
+                "error": f"Failed to list pull requests: {response.text}",
+                "status_code": response.status_code,
+            }
+
```

Comment:
Would it be worth making a `GitHubClient` class? Can come in a followup, just conscious there's a ton of GitHub specific code in an otherwise unrelated file

Reply:
Or maybe even using something like https://github.com/PyGithub/PyGithub